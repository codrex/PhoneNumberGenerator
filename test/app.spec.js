const request = require('supertest');
const app = require('../app');

describe('App', () => {
  describe('GET / ', () => {
    it('respond with html string', (done) => {
      request(app)
        .get('/')
        .set('Accept', 'application/json')
        .expect('Content-Type', 'text/html; charset=utf-8')
        .expect(200)
        .end((err, res) => {
          expect(res.text).toMatchSnapshot();
          done();
        });
    });
  });

  describe('GET /sort/:order ', () => {
    it('respond with html string', (done) => {
      request(app)
        .get('/sort/ASC')
        .set('Accept', 'application/json')
        .expect('Content-Type', 'text/html; charset=utf-8')
        .expect(200, done);
    });
  });

  describe('POST /generate/:number ', () => {
    it('respond with html string', (done) => {
      request(app)
        .post('/generate/100')
        .set('Accept', 'application/json')
        .expect('Content-Type', 'text/html; charset=utf-8')
        .expect(200, done);
    });
  });

  describe('404 route ', () => {
    it('respond with html string', (done) => {
      request(app)
        .post('/unknown')
        .set('Accept', 'application/json')
        .expect('Content-Type', 'text/html; charset=utf-8')
        .expect(200)
        .end((err, res) => {
          expect(res.text).toMatchSnapshot();
          done();
        });
    });
  });
});
