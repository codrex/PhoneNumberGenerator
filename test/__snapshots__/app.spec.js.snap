// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App 404 route  respond with html string 1`] = `"<!DOCTYPE html><html><head><title></title><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\"><link rel=\\"stylesheet\\" href=\\"https://fonts.googleapis.com/css?family=Nunito\\"><link rel=\\"stylesheet\\" href=\\"https://use.fontawesome.com/e42a9be8b9.css\\" media=\\"all\\"><link rel=\\"stylesheet\\" href=\\"https://stackpath.bootstrapcdn.com/bootstrap/4.1.3/css/bootstrap.min.css\\" integrity=\\"sha384-MCw98/SFnGE8fJT3GXwEOngsV7Zt27NXFoaoApmYm81iuXoPkFOJwJ8ERdknLPMO\\" crossorigin=\\"anonymous\\"><link rel=\\"stylesheet\\" href=\\"stylesheets/style.css\\"><script src=\\"javascripts/index.min.js\\"></script></head><body><div class=\\"toast\\"><p class=\\"toast-text\\">404: Page not found</p></div><div class=\\"error\\"><h1>Oops!</h1><p>404: Page not found</p></div></body></html>"`;

exports[`App GET /  respond with html string 1`] = `"<!DOCTYPE html><html><head><title>phone number generator</title><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\"><link rel=\\"stylesheet\\" href=\\"https://fonts.googleapis.com/css?family=Nunito\\"><link rel=\\"stylesheet\\" href=\\"https://use.fontawesome.com/e42a9be8b9.css\\" media=\\"all\\"><link rel=\\"stylesheet\\" href=\\"https://stackpath.bootstrapcdn.com/bootstrap/4.1.3/css/bootstrap.min.css\\" integrity=\\"sha384-MCw98/SFnGE8fJT3GXwEOngsV7Zt27NXFoaoApmYm81iuXoPkFOJwJ8ERdknLPMO\\" crossorigin=\\"anonymous\\"><link rel=\\"stylesheet\\" href=\\"stylesheets/style.css\\"><script src=\\"javascripts/index.min.js\\"></script></head><body><div class=\\"toast\\"><p class=\\"toast-text\\"></p></div><div class=\\"content\\"><h1 class=\\"title\\">Phone Number Generator</h1><form class=\\"form\\"><div class=\\"form-group\\"><button class=\\"c-btn c-btn--primary\\" type=\\"submit\\">Generate numbers</button><input class=\\"num-input\\" type=\\"number\\"></div><p class=\\"help-text\\">Enter a number between 1 and 500</p></form><div class=\\"table-container\\"><div class=\\"table-header\\"><p class=\\"table-header-text\\">total:  4</p><p class=\\"table-header-text\\">max number:  5050505050</p><p class=\\"table-header-text\\">min number:  0940595069</p></div><div class=\\"table-responsive\\"><table class=\\"table table-striped table-borderless table-hover\\"><thead><tr><th scope=\\"col\\">#</th><th scope=\\"col\\" id=\\"phone-number\\">Phone number <i class=\\"fa fa-sort sort-icon\\" aria-hidden=\\"true\\" role=\\"presentation\\"></i></th><th scope=\\"col\\">assigned</th></tr></thead><tbody><tr><th scope=\\"row\\">1<td>0940595069</td><td>false</td></th></tr><tr><th scope=\\"row\\">2<td>22222222</td><td>false</td></th></tr><tr><th scope=\\"row\\">3<td>2222cc2222</td><td>false</td></th></tr><tr><th scope=\\"row\\">4<td>5050505050</td><td>false</td></th></tr></tbody></table></div></div></div></body></html>"`;
